sdks/java/core/src/test/java/org/apache/beam/sdk/io/WriteTest.java

```diff
@@ -54,14 +62,55 @@
 import java.util.Objects;
 import java.util.Set;
 import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
 
 /**
  * Tests for the Write PTransform.
  */
 @RunWith(JUnit4.class)
 public class WriteTest {
   // Static store that can be accessed within the writer
-  static List<String> sinkContents = new ArrayList<>();
+  private static List<String> sinkContents = new ArrayList<>();
+
+  private static final MapElements<String, String> IDENTITY_MAP =
+      MapElements.via(new SimpleFunction<String, String>() {
+        @Override
+        public String apply(String input) {
+          return input;
+        }
+      });
+
+  private static class WindowAndReshuffle<T> extends PTransform<PCollection<T>, PCollection<T>> {
+    private final Window.Bound<T> window;
+    public WindowAndReshuffle(Window.Bound<T> window) {
+      this.window = window;
+    }
+
+    private static class AddArbitraryKey<T> extends DoFn<T, KV<Integer, T>> {
+      @Override
+      public void processElement(ProcessContext c) throws Exception {
+        c.output(KV.of(ThreadLocalRandom.current().nextInt(), c.element()));
+      }
+    }
+
+    private static class RemoveArbitraryKey<T> extends DoFn<KV<Integer, Iterable<T>>, T> {
+      @Override
+      public void processElement(ProcessContext c) throws Exception {
+        for (T s : c.element().getValue()) {
+          c.output(s);
+        }
+      }
+    }
+
+    @Override
+    public PCollection<T> apply(PCollection<T> input) {
+      return input
+          .apply(window)
+          .apply(ParDo.of(new AddArbitraryKey<T>()))
+          .apply(GroupByKey.<Integer, T>create())
+          .apply(ParDo.of(new RemoveArbitraryKey<T>()));
```

Comment:
Up to you, but you can use Values.create() and Flatten.flattenIterables() instead of RemoveArbitraryKey